audiophiler/__init__.py

```diff
@@ -225,25 +263,45 @@ def get_harold(uid, auth_dict=None):
 
                 return get_file_s3(s3_bucket, harold_file_hash)
 
+    return "Permission denied", 403
+
+@app.route("/lock", method=["POST"])
+@auth.oidc_auth('default')
+@audiophiler_auth
+def toggle_tour_mode(auth_dict=None):
+    is_rtp = ldap_is_rtp(auth_dict["uid"])
+    is_eboard = ldap_is_eboard(auth_dict["uid"])
+    if not is_rtp or not is_eboard:
+        return "Permisssion Denied", 403
 
+    admin_query = Tour.query.first()
+    admin_query.tour_lock = not admin_query.tour_lock
 
-    return "Permission denied", 403
+    db.session.flush()
+    db.session.commit()
+
+    return "Tour Mode toggled", 200
 
 @app.route("/logout")
 @auth.oidc_logout
 def logout():
     return redirect("/", 302)
 
 def get_harold_list(uid):
-    harold_list = Harold.query.filter_by(owner=uid).all()
+    local_uid = uid
+    if get_tour_lock_status():
+        local_uid ="root"
```

Comment:
So this will show only tour mode selections if the application is in tour mode, regardless of the user. What if I want to manage my personal harolds during an open house?

Why not check if this is being accessed from the tour mode management page?